### PR TITLE
chore: Add a precompile query test  and fix avs cli

### PR DESCRIPTION
**Pull request by trestinlsd**
## Description

----

Add a precompile query test  and fix avs cli

## Summary by CodeRabbit

- **New Features**
	- Introduced new methods for retrieving operator public keys and opted-in operators in the AVS precompile contract.
	- Added a new command for querying task information with improved argument structure and error messages.
	- Enhanced the `OperatorKeeper` interface with a method to retrieve opted-in operators by AVS address.

- **Bug Fixes**
	- Improved error handling and messages for invalid addresses in command queries.

- **Refactor**
	- Removed deprecated methods related to operator queries from the precompile structure.

- **Tests**
	- Added comprehensive unit tests for the AVS precompile functionalities to ensure reliability.

### CODE REVIEW

**Review by mikebraver — APPROVED** [review]

File: x/avs/client/cli/query.go [hunk 1]
```diff
@@ -90,16 +90,15 @@ func QueryAVSAddrByChainID() *cobra.Command {
...
+func QueryTaskInfo() *cobra.Command {
 	cmd := &cobra.Command{
-		Use:   "GetTaskInfo avstask info",
-		Short: "Get avstask info",
-		Long:  "Get avstask info",
-		Args:  cobra.ExactArgs(1),
+		Use:   "TaskInfo query",
+		Short: "TaskInfo query",
+		Long:  "TaskInfo query for current registered task",
+		Args:  cobra.ExactArgs(2),
```
Comment (MaxMustermann2) [inline x/avs/client/cli/query.go]:
```suggestion
		Use:   "TaskInfo <task-address-in-hex> <task-id>",
		Short: "Query the TaskInfo by its address and ID",
		Long:  "Query the currently registered tasks for an AVS by the task's address and ID",
		Args:  cobra.ExactArgs(2),
```

File: x/avs/client/cli/query.go [hunk 2]
```diff
@@ -109,6 +108,7 @@ func GetTaskInfo() *cobra.Command {
 			queryClient := types.NewQueryClient(clientCtx)
 			req := types.QueryAVSTaskInfoReq{
 				TaskAddr: args[0],
+				TaskId:   args[1],
```
Comment (MaxMustermann2) [inline x/avs/client/cli/query.go:111]: Does this need any validation?

File: x/avs/types/expected_keepers.go [hunk 3]
```diff
@@ -37,6 +37,7 @@ type OperatorKeeper interface {
 	IsOperator(ctx sdk.Context, addr sdk.AccAddress) bool
 	OptIn(ctx sdk.Context, operatorAddress sdk.AccAddress, avsAddr string) error
 	OptOut(ctx sdk.Context, operatorAddress sdk.AccAddress, avsAddr string) (err error)
+	GetOptedInOperatorListByAVS(ctx sdk.Context, avsAddr string) ([]string, error)
```
Comment (MaxMustermann2) [inline x/avs/types/expected_keepers.go:40]: Will you implement this?

File: x/avs/client/cli/query.go (hunk 1 above)
Comment (trestinlsd) [reply x/avs/client/cli/query.go]: fixed

File: x/avs/client/cli/query.go (hunk 2 above)
Comment (trestinlsd) [reply x/avs/client/cli/query.go:111]: Simple numerical values, currently do not require verification

File: x/avs/types/expected_keepers.go (hunk 3 above)
Comment (trestinlsd) [reply x/avs/types/expected_keepers.go:40]: Already implemented, on line 197 of the x/operator/keeper/operator.go file

**Review by MaxMustermann2 — APPROVED** [review]

**Review by TimmyExogenous — APPROVED** [review]

**Pull request merged**